e2e-tests/functions

```diff
@@ -87,3 +89,118 @@ get_psql_user_host() {
 
 	kubectl -n ${NAMESPACE} get "secret/${cluster}-pguser-${user}" -o jsonpath='{.data.host}' | base64 -d
 }
+
+wait_pod() {
+	local pod=$1
+
+	set +o xtrace
+	retry=0
+	echo -n $pod
+	until kubectl get pod/$pod -n "${NAMESPACE}" -o jsonpath='{.status.containerStatuses[0].ready}' 2>/dev/null | grep 'true'; do
+		sleep 1
+		echo -n .
+		let retry+=1
+		if [ $retry -ge 360 ]; then
+			kubectl describe pod/$pod -n "${NAMESPACE}"
+			kubectl logs $pod -n "${NAMESPACE}"
+			kubectl logs $(get_operator_pod) -n "${NAMESPACE}" \
+				| grep -v 'level=info' \
+				| grep -v 'level=debug' \
+				| grep -v 'Getting tasks for pod' \
+				| grep -v 'Getting pods from source' \
+				| tail -100
+			echo max retry count $retry reached. something went wrong with operator or kubernetes cluster
+			exit 1
+		fi
+	done
+	set -o xtrace
+}
+
+get_service_ip() {
+	local service=$1
+	while (kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.spec.type}' 2>&1 || :) | grep -q NotFound; do
+		sleep 1
+	done
+	if [ "$(kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.spec.type}')" = "ClusterIP" ]; then
+		kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.spec.clusterIP}'
+		return
+	fi
+	until kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.status.loadBalancer.ingress[]}' 2>&1 | egrep -q "hostname|ip"; do
+		sleep 1
+	done
+	kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.status.loadBalancer.ingress[].ip}'
+	kubectl get service/$service -n "${NAMESPACE}" -o 'jsonpath={.status.loadBalancer.ingress[].hostname}'
+}
+
+deploy_pmm_server() {
+	helm uninstall -n "${NAMESPACE}" pmm || :
+	kubectl delete clusterrolebindings.rbac.authorization.k8s.io pmm || :
+	kubectl delete clusterrole.rbac.authorization.k8s.io pmm || :
+	helm repo remove percona || :
+	helm repo add percona https://percona.github.io/percona-helm-charts/
+	helm install -n "${NAMESPACE}" pmm percona/pmm --set service.type="LoadBalancer"
+	wait_pod pmm-0
+}
+
+generate_pmm_api_key() {
```

Comment:
Maybe rename it to `generate_pmm_server_key` just for consistency reasons since we use `PMM_SERVER_KEY`